src/cs/types_visitor.ts

```diff
@@ -0,0 +1,92 @@
+import { Argument, BaseVisitor, Context } from "@apexlang/core/model";
+import { camelCase, formatComment, pascalCase } from "../utils";
+import { expandType } from "./helpers";
+
+export class TypeVisitor extends BaseVisitor {
+  visitTypeBefore(context: Context): void {
+    const { type } = context;
+
+    this.write(formatComment("/// ", type.description));
+    this.write(`class ${pascalCase(type.name)}\n{\n`);
```

Comment:
Could you look into using records here instead of classes? I believe they simplify the generated code among other advantages.

Reply:
@pkedy i have completed this and will push in some time.

Do you have a plan on how to implement the services, I mean should i go with the functional pattern like the .MapGet etc or should i implement the controller pattern?

Reply:
Services as in REST/HTTP services? Seems like .NET minimal APIs are pretty exciting (e.g. MapGet).

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-6.0